deployer/src/deployment/deploy_layer.rs

```diff
@@ -574,9 +574,14 @@ mod tests {
     async fn test_states(id: &Uuid, expected_states: Vec<StateLog>) {
         loop {
             let states = RECORDER.lock().unwrap().get_deployment_states(id);
+            if states == expected_states {
+                return;
+            }
 
-            if *states == expected_states {
-                break;
+            for (actual, expected) in states.iter().zip(&expected_states) {
+                if actual != expected {
+                    return;
+                }
```

Comment:
Do we really need this?

Reply:
When the list of state diverges from the expected states (e.g. the runtime crashes), if we don't return early the test is going to run until the timeout is reached even though we know that it can never be the same. When trying to reproduce the bug, I sometimes had to wait for several minutes for the timeout to trigger and the test(s) to end.

Reply:
To be able to stop the loop early ! The first `if` statement only covers the case where every element is the same. But while the deployer is running, this isn't the case, hence the loop and the sleep statement. If don't do the for loop though, we have to rely on the timeout for the future to be canceled, even though we might have known earlier that the timeout that the states could never be the same (e.g. if we have the expected states `Build, Running, Completed`, and the actual state `Build, Crashed`), we know without having to wait that the exit condition will never happen).